src/main/scala/org/broadinstitute/dsde/vault/model/Generic.scala

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.dsde.vault.model
+
+import com.wordnik.swagger.annotations.{ApiModel, ApiModelProperty}
+import spray.json.DefaultJsonProtocol
+import scala.annotation.meta.field
+
+object GenericJsonProtocol extends DefaultJsonProtocol {
+  implicit val impGenericSysAttrs = jsonFormat5(GenericSysAttrs)
+  implicit val impGenericEntity = jsonFormat6(GenericEntity)
+  implicit val impGenericRelationship = jsonFormat2(GenericRelationship)
+  implicit val impGenericRelEnt = jsonFormat2(GenericRelEnt)
+  implicit val impBlobSpecification = jsonFormat2(BlobSpecification)
+  implicit val impGenericEntityIngest = jsonFormat4(GenericEntityIngest)
+  implicit val impGenericRelationshipIngest = jsonFormat4(GenericRelationshipIngest)
+  implicit val impGenericIngest = jsonFormat2(GenericIngest)
+  implicit val impGenericAttributeSpec = jsonFormat2(GenericAttributeSpec)
+  implicit val impGenericQuery = jsonFormat3(GenericEntityQuery)
+}
+
+@ApiModel("system-generated entity attributes")
+case class GenericSysAttrs(
+  @(ApiModelProperty@field)("optional BOSS ID")
+  bossID: Option[String],
+  @(ApiModelProperty@field)("when the entity was created (mSecs since epoch)")
+  createdDate: Long,
+  @(ApiModelProperty@field)("openAM commonName of entity creator")
+  createdBy: String,
+  @(ApiModelProperty@field)(value="when the entity was most recently modified (mSecs since epoch)",dataType="Long")
+  modifiedDate: Option[Long],
+  @(ApiModelProperty@field)("openAM commonName of most recent modifier")
+  modifiedBy: Option[String] )
+
+@ApiModel("something that lives in the vault")
+case class GenericEntity(
+  @(ApiModelProperty@field)("the globally unique vault ID for the entity")
+  guid: String,
+  @(ApiModelProperty@field)("a signed PUT URL for time-limited ingest of BOSS objects")
+  putUrl: Option[String],
+  @(ApiModelProperty@field)("a signed GET URL for time-limited retrieval of BOSS objects")
+  getUrl: Option[String],
```

Comment:
huh. There appears to be some swagger magic that translates the "getUrl" method to "url" when it is describing the response model/model schema. I think we explicitly want the response to contain "getUrl" to differentiate it from "putUrl".

I suppose getUrl/putUrl is another hurdle to sharing models in vault-common.